xkb/xkb.c

```diff
@@ -6065,7 +6065,9 @@ ProcXkbGetKbdByName(ClientPtr client)
         }
 
         memcpy(payload_walk, &mrep, sizeof(mrep));
-        payload_walk = buf + (mrep.length * 4) - (sizeof(mrep) - sizeof(xGenericReply));
+
+        // length is in 4 byte units and only accounts what's above sizeof(xGenericReply)
+        payload_walk += mrep.length * sizeof(CARD32) + sizeof(xGenericReply);
```

Comment:
Should there be an opposite to `bytes_to_int32()` (e.g. `int32_to_bytes()`) for this? I don't really like the "int32" part of the name (OTOH I'm having a hard time finding X11 spec calling it anything but "4-byte units" and "units of four bytes"; should at least have been plural "int32s" if not something like "dwords"), but otherwise might be a good idea…

It could also be a macro like e.g.
```c
#define REPLY_SIZE(r) ((r).length * sizeof(CARD32) + sizeof(xGenericReply))
// …
payload_walk += REPLY_SIZE(mrep);
```

Reply:
hmm, or chunks ? pads? ...

Reply:
posted a survey on mailing list. let's see whether somebody coming up with a really good name.